src/main/java/org/opentripplanner/routing/impl/OrcaFareServiceImpl.java

```diff
@@ -0,0 +1,251 @@
+            } else if (routeData.getType() == 4 && routeData.getDesc().contains("Water Taxi: Vashon Island")) {
+                return RideType.KC_WATER_TAXI_VASHON_ISLAND;
+            }
+            return RideType.KC_METRO;
+        } else if ("40".equals(routeData.getAgency().getId())) {
+            return RideType.SOUND_TRANSIT;
+        } else if ("97".equals(routeData.getAgency().getId())) {
+            return RideType.EVERETT_TRANSIT;
+        } else if ("3".equals(routeData.getAgency().getId())) {
+            return RideType.PIERCE_COUNTY_TRANSIT;
+        } else if ("23".equals(routeData.getAgency().getId())) {
+            return RideType.SEATTLE_STREET_CAR;
+        } else if ("wsf".equalsIgnoreCase(routeData.getAgency().getId())) {
+            return RideType.WASHINGTON_STATE_FERRIES;
+        } else if ("kt".equalsIgnoreCase(routeData.getAgency().getId())) {
+            return RideType.KITSAP_TRANSIT;
+        }
+        return null;
+    }
+
+    /**
+     * Define which discount fare should be applied based on the fare type. If the ride type is unknown the discount
+     * fare can not be applied, use the default fare.
+     */
+    private float getDiscountedFare(Fare.FareType fareType, RideType rideType, float defaultFare) {
+        if (rideType == null) {
+            return defaultFare;
+        }
+        switch (fareType) {
+            case youth:
+            case orcaYouth:
+                return getYouthFare(rideType);
+            case orcaLift:
+                return getLiftFare(rideType, defaultFare);
+            case orcaSenior:
+            case senior:
+                return getSeniorFare(fareType, rideType, defaultFare);
+            case orcaRegular:
+                return getRegularFare(rideType, defaultFare);
+            case regular:
+            default:
+                return defaultFare;
+        }
+    }
+
+    /**
+     * Apply Orca regular discount fares. If the ride type can not be matched the default fare is used.
+     */
+    private float getRegularFare(RideType rideType, float defaultFare) {
+        switch (rideType) {
+            case KC_WATER_TAXI_VASHON_ISLAND: return 5.75f;
+            case KC_WATER_TAXI_WEST_SEATTLE: return 5.00f;
+            default: return defaultFare;
+        }
+    }
+
+    /**
+     * Apply Orca lift discount fares based on the ride type.
+     */
+    private float getLiftFare(RideType rideType, float defaultFare) {
+        switch (rideType) {
+            case COMM_TRANS_LOCAL_SWIFT: return 1.25f;
+            case COMM_TRANS_COMMUTER_EXPRESS: return 2.00f;
+            case PIERCE_COUNTY_TRANSIT: return defaultFare;
+            case KC_WATER_TAXI_VASHON_ISLAND: return 4.50f;
+            case KC_WATER_TAXI_WEST_SEATTLE: return 3.75f;
+            case KITSAP_TRANSIT: return 1.00f;
+            // KCM, Sound Transit, Everett, Seattle Streetcar.
+            default: return 1.50f;
+        }
```

Comment:
Seems like the default case should return the default fare.

```suggestion
        switch (rideType) {
            case COMM_TRANS_LOCAL_SWIFT: return 1.25f;
            case COMM_TRANS_COMMUTER_EXPRESS: return 2.00f;
            case KC_WATER_TAXI_VASHON_ISLAND: return 4.50f;
            case KC_WATER_TAXI_WEST_SEATTLE: return 3.75f;
            case KITSAP_TRANSIT: return 1.00f;
            case KC_METRO:
            case SOUND_TRANSIT:
            case EVERETT_TRANSIT:
            case SEATTLE_STREET_CAR:
                return 1.50f;
            case PIERCE_COUNTY_TRANSIT: 
            default:
                return defaultFare;
        }
```